### PR TITLE
[wolfSSL] update version to v5.4.0 AND [wolfTPM] update version to v2.5.0 AND [wolfMQTT] update version to v1.14.0

### LINKED ISSUE #25858: [wolfSSL] update to 5.4.0
Library name: wolfSSL

New version number: 5.4.0

Other information that may be useful (release notes, etc...)
https://github.com/wolfSSL/wolfssl/releases

### PR DESCRIPTION
**Pull request by dgarske**
- #### What does your PR fix?
Updated to stable wolfSSL v5.4.0
Updated to stable wolfTPM v2.5.0
Updated to stable wolfMQTT v1.14.0
Fixes #25858
- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
yes

### CODE REVIEW

**Comment by dgarske** [comment]
FYI: @JacobBarthelmeh and @embhorn

**Comment by FrankXie05** [comment]
@dgarske Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).
Also please run command `vcpkg x-add-version --all` then commit the changes.

**Comment by dgarske** [comment]
The failures are due to a wolfSSL CMake bug with v5.4.0 where our SP math all is enabled by default, yet the wolfssl/wolfcrypt/sp_int.h is not being installed. Patch here: https://github.com/wolfSSL/wolfssl/pull/5401
I'll be updating with a post release patch for wolfSSL.

**Comment by dgarske** [comment]
Hi @FrankXie05 , having trouble reproducing the latest errors. I have only tried on Mac OS X and Ubuntu 20. Any tips?

**Comment by FrankXie05** [comment]
@dgarske
Log from `install-x86-windows-dbg-out.log`  of `wolfmqtt`
````
D:\installed\x86-windows\include\wolfssl/wolfcrypt/integer.h(37): fatal error C1083: Cannot open include file: 'wolfssl/wolfcrypt/sp_int.h': No such file or directory
ninja: build stopped: subcommand failed.
````
Regarding this issue, I see that you have fixed it upstream, but the current commit doesn't merge your content, we need a patch until the next release

Log from `install-x64-windows-dbg-out.log` of `wolftpm`
````
D:\buildtrees\wolftpm\src\v2.5.0-942afdea65.clean\wolftpm/options.h(60): fatal error C1070: mismatched #if/#endif pair in file 'D:\buildtrees\wolftpm\src\v2.5.0-942afdea65.clean\wolftpm\options.h'
ninja: build stopped: subcommand failed.
````
The problem is that the file `options.h` does not have permission to read and write content(https://github.com/wolfSSL/wolfTPM/blob/d7b76cdee52715ded9066374432428528d3b22e2/CMakeLists.txt#L201), resulting in `#if` not matching `#endif`.
````
CMake Error at CMakeLists.txt:211 (file):
  file failed to open for writing (Permission denied):

    D:/buildtrees/wolftpm/src/v2.5.0-942afdea65.clean/wolftpm/options.h
````
[config-x86-windows-out (3).log](https://github.com/microsoft/vcpkg/files/9194571/config-x86-windows-out.3.log)
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/9194577/install-x64-windows-dbg-out.log)
[install-x86-windows-dbg-out (1).log](https://github.com/microsoft/vcpkg/files/9194579/install-x86-windows-dbg-out.1.log)

**Comment by dgarske** [comment]
@FrankXie05 this is good to go now. Thanks for your help.

File: ports/wolfmqtt/portfile.cmake [hunk 1]
```diff
@@ -1,9 +1,11 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO wolfssl/wolfmqtt
-    REF v1.13.0
-    SHA512 2bd05471c548bacf6fc5166d9c2ee546b3ce5c4fab89ff16175de9ab7696b8b570cdd7d4377378fc0cb579447488dbf727cb938192eb3180ed007414cb949a8a
+    REF v1.14.0
+    SHA512 9449d87c543e823b3517a4605343a92207499812e75c950a2e8fb3d969333d39579dac69657e298826ba65017a8208c28934d7c6a4abbb4bf308514047273191
     HEAD_REF master
+    PATCHES
+      wolfmqtt_pr305.diff
```
Comment (FrankXie05) [inline ports/wolfmqtt/portfile.cmake:8]:
```suggestion
      wolfmqtt_pr305.patch
```
Sorry for the late reply. Please change the patch to the unified naming format `xx.patch`. :)

File: ports/wolfssl/portfile.cmake [hunk 2]
```diff
@@ -1,9 +1,11 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO wolfssl/wolfssl
-    REF v5.3.0-stable
-    SHA512 399d2b8aad58471d237d21dea68c33fde2b9a3c117c554c241d9174db02847a6c31afae2908839d18f9ada317b2388560a24c077b76014f663227061342bf045
+    REF v5.4.0-stable
+    SHA512 e43560f83f6c62d78b10c4df7db21f02386f22b893688b98d2f3432e3b3946a4e80960c7402404a8c0486c87b1dde5b7a3827f9d4d3be13f87f370dfc1179c78
     HEAD_REF master
+    PATCHES
+      wolfssl_pr5401.diff
```
Comment (FrankXie05) [inline ports/wolfssl/portfile.cmake:8]:
```suggestion
      wolfssl_pr5401.patch
```

File: ports/wolftpm/portfile.cmake [hunk 3]
```diff
@@ -1,14 +1,17 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO wolfssl/wolftpm
-    REF v2.4.0
-    SHA512 82aa43f8773cc230d403d7e6cdbc4dc7dc8a3cb8e2f942c810a5ef0d60647d288ce82910820127ed6e67771d3900cfc55c5f9d87b722e0847bcb6a81c4e41441
+    REF v2.5.0
+    SHA512 4047671171fda4115a4e94bbd2362642752efbaf30531b1f55f6fe66e2e90bdf3b5549c6ddf4e53dce3b6f44c59e792217b2be4c5650857d28193aa5466919fc
     HEAD_REF master
+    PATCHES
+      wolftpm_pr235.diff
```
Comment (FrankXie05) [inline ports/wolftpm/portfile.cmake:8]:
```suggestion
      wolftpm_pr235.patch
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/wolfmqtt/portfile.cmake (hunk 1 above)
Comment (LilyWangLL) [reply ports/wolfmqtt/portfile.cmake:8]: This does not need to be changed, `.diff` is correct.

**Review by LilyWangLL — APPROVED** [review]

**Review by FrankXie05 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Looks great, thanks for the contribution!

**Pull request merged**